File: leetcode_autopush/3799-unique-3-digit-even-numbers/unique-3-digit-even-numbers.cpp
class Solution {
public:
    int totalNumbers(vector<int>& digits) {
        unordered_set<int> unique;
        int len= digits.size();
        for(int i=0;i<len;i++){
            if(digits[i]==0){
                continue;
            }

            for(int j=0;j<len;j++){
                if(j==i){
                    continue;
                }

                for(int k=0;k<len;k++){
                    if(k==i||k==j){
                        continue;
                    }

                    int one = digits[k];
                    if(one%2!=0){
                        continue;
                    }
                    int hund = digits[i];
                    int ten = digits[j];
                    int num = hund*100 + ten*10+one;
                    unique.insert(num);


                }
            }
        }
        return unique.size();
    }
};

File: leetcode_autopush/3799-unique-3-digit-even-numbers/unique-3-digit-even-numbers.java
class Solution {
    public int totalNumbers(int[] digits) {
        int[] availablecount = new int[10];

        for(int d: digits){
            availablecount[d]++;
        }

        int validcount =0;

        for(int num=100;num<1000;num+=2){
            int hund = num/100;
            int ten = (num/10)%10;
            int one = num%10;

            int[] needcount = new int[10];
            needcount[hund]++;
            needcount[ten]++;
            needcount[one]++;

            boolean valid = true;
            for(int i=0;i<10;i++){
                if(availablecount[i]<needcount[i]){
                    valid = false;
                    break;
                }
            }

            if(valid){
                validcount++;
            }
        }

        return validcount;

    }
}